cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -209,10 +217,12 @@ public Future<KafkaStatus> createOrUpdate(Reconciliation reconciliation, Kafka k
     Future<Void> reconcile(ReconciliationState reconcileState)  {
         Promise<Void> chainPromise = Promise.promise();
 
-        boolean isKRaftEnabled = featureGates.useKRaftEnabled() && ReconcilerUtils.kraftEnabled(reconcileState.kafkaAssembly);
+        KafkaMetadataConfigurationState kafkaMetadataConfigState = reconcileState.kafkaMetadataStateManager.getMetadataConfigurationState();
+        // since PRE_MIGRATION phase (because it's when controllers are deployed during migration) we need to validate usage of node pools and features for KRaft
+        LOGGER.infoCr(reconcileState.reconciliation, "KafkaMetadataConfigurationState = {}, isPreMigrationOrKRaft = {}", kafkaMetadataConfigState, kafkaMetadataConfigState.isPreMigrationOrKRaft());
```

Comment:
Leftover log message from development?

Reply:
Yes, I will review all INFO level logging when we are close to accept the PR. They are still useful to me for development/testing during the review as well.